fabric-events-player-v1/src/main/java/net/fabricmc/api/event/player/v1/PlayerRespawnCallback.java

```diff
@@ -0,0 +1,34 @@
+package net.fabricmc.api.event.player.v1;
+
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.dimension.DimensionType;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Represents a callback for when a {@link ServerPlayerEntity} is respawned.
+ *
+ * <p>This occurs when a player dies, or changes dimensions, such as returning to the overworld from the end.
+ *
+ * <p>Note the player can be repositioned.
+ */
+public interface PlayerRespawnCallback {
+	Event<PlayerRespawnCallback> EVENT = EventFactory.createArrayBacked(PlayerRespawnCallback.class, (callbacks) -> (newPlayer, oldPlayer, newDimension, alive) -> {
+		for (PlayerRespawnCallback callback : callbacks) {
+			callback.onRespawn(newPlayer, oldPlayer, newDimension, alive);
+		}
+	});
+
+	/**
+	 * Called when a player respawns.
+	 *
+	 * <p>Note the player can be repositioned.
+	 *
+	 * @param newPlayer The new {@link ServerPlayerEntity} that will be spawned.
+	 * @param oldPlayer The old {@link ServerPlayerEntity} that is being removed.
+	 * @param dimension The dimension this player is being respawned in.
+	 * @param oldPlayerIsAlive Whether the old player is still alive.
+	 */
+	void onRespawn(ServerPlayerEntity newPlayer, ServerPlayerEntity oldPlayer, DimensionType dimension, boolean oldPlayerIsAlive);
```

Comment:
Oh sorry, I didn't mean the parameter should be renamed. `isAlive` was fine, it just needed clarification in the doc. `oldPlayerIsAlive` is a tad verbose I think.